ODIN_II/SRC/odin_ii.cpp

```diff
@@ -183,14 +199,50 @@ static void optimization() {
     printf("\n--------------------------------------------------------------------\n");
 }
 
+/*---------------------------------------------------------------------------
+ * (function: get_current_dir)
+ returns current working directory
+ *-------------------------------------------------------------------------*/
+std::string get_current_dir() {
+   char path_buffer[FILENAME_MAX]; //create string buffer for path
+   GetCurrentDirectory( path_buffer, FILENAME_MAX );
+   string curr_working_dir(path_buffer);
+   return curr_working_dir;
+}
+
+/*---------------------------------------------------------------------------
+ * (function: predict_optimal_ratio)
+ returns optimal ratio for the current combination of circuit and architecture file
+ *-------------------------------------------------------------------------*/
+float predict_optimal_ratio(){
```

Comment:
This function must not be here in odin_ii.cpp
if you used mixing optimization, why didn't you add these predict optimal under the same subsystem?